src/net_processing.cpp

```diff
@@ -2877,15 +2983,30 @@ void ProcessMessage(
                 LogPrint(BCLog::MEMPOOL, "not keeping orphan with rejected parents %s\n",tx.GetHash().ToString());
                 // We will continue to reject this tx since it has rejected
                 // parents so avoid re-requesting it from other peers.
+                // Here we add both the txid and the wtxid, as we know that
+                // regardless of what witness is provided, we will not accept
+                // this, so we don't need to allow for redownload of this txid
+                // from any of our non-wtxidrelay peers.
                 recentRejects->insert(tx.GetHash());
```

Comment:
I'm not very familiar with alternative (deployed) SIGHASH flags, but I was wondering if this can possible screw us somehow... I don't have a particular attack scenario, just throwing this idea here.

Reply:
I don't know what SIGHASH flags has to do with this, can you elaborate on the scenario you have in mind?

My understanding of the logic here is that at this point in the code, we know that a parent of this orphan transaction will definitely not be accepted by our node (as it is already in our reject filter).  Since a txid is a commitment to the inputs of a transaction, this means that we can definitively know that we'll never accept any transaction with the same txid as this orphan, regardless of what witness it comes with.

Reply:
Nevermind, I was worried that something would happen to xxx | ANYONECANPAY, but now I think there are no issues there.